src/main/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -967,11 +967,21 @@ private boolean getSecondOfPairFlagUnchecked() {
 
     /**
      * the alignment is not primary (a read having split hits may have multiple primary alignment records).
+     * @deprecated use {@link #getSecondaryAlignmentFlag()} instead.
      */
+    @Deprecated
     public boolean getNotPrimaryAlignmentFlag() {
-        return (mFlags & SAMFlag.NOT_PRIMARY_ALIGNMENT.flag) != 0;
+        return getSecondaryAlignmentFlag();
+    }
+
+    /**
+     * the alignment is secondary (a read having slipt hits have multiple alignment records).
+     */
+    public boolean getSecondaryAlignmentFlag() {
```

Comment:
time to switch to 

`isSecondaryAlignmentFlag() `

?

Reply:
I would say yes, but there is another PR for discussing that: https://github.com/samtools/htsjdk/pull/868

Reply:
Or even `isSecondaryAlignment()`

Reply:
I agree, but wouldn't be discordant with the rest of the class? I guess that in the PR that I mentioned it was decided to change everything together... but I added a commit changing that.